### PR TITLE
Add NPC Comment Blocker - Block annoying spam like comments on youtube

### PR DESCRIPTION
**Pull request by whirlxd**
### Project Details
1. **What is the name of your project?** - NPC Comment Blocker

3. **How many hours did you spend developing this?**  - 8-10

4. **What problem does your extension solve, or what makes it unique?**  
There are a lot of spam / npc like comments on youtube like - "whose here in 2024 , only legends like this , etc.. " so as someone who likes to browse comments and guage a sense of if the video is worth watching and some points about the content in it these comments become very annoying. One may argue that you could program a taper monkey script however it is aimed at non programmers aswell and due to the number of people using it and recommending it i have been able to grow the list of common keywords in all of these spam comments

5. **How many people have tested your extension?**  
Personally with friends and family over 7 people have and i posted it on reddit and some discord servers so i believe over atleast 50 people have used / tested it

6. **Does your extension work in both Chrome and Firefox?**  
Yes it does!

---

### CODE REVIEW

**Review by JosiasAurel — CHANGES_REQUESTED** [review]
Please answer the questions. I also get this error when trying to load the extension. 
image

**Comment by whirlxd** [comment]
hey , i uploaded the code and i believe i missed the icons part , ill fix that really sorry for that

**Comment by whirlxd** [comment]
@JosiasAurel fixed!

**Comment by JosiasAurel** [comment]
Can you link an example video where this works?

**Comment by whirlxd** [comment]
https://cloud-asju9zw95-hack-club-bot.vercel.app/020241118-2017-32.4046972.mp4

**Comment by JosiasAurel** [comment]
> https://cloud-asju9zw95-hack-club-bot.vercel.app/020241118-2017-32.4046972.mp4

i meant the youtube link so i can try this myself. at least 3 please

**Comment by whirlxd** [comment]
@JosiasAurel

**Comment by whirlxd** [comment]
hey @JosiasAurel

File: submission/npc-comment-blocker/content.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
+const keywords = [
```
Comment (JosiasAurel) [inline submission/npc-comment-blocker/content.js:1]: This doesn't seem like a reliable way of identifying bot comments. You could have a [model trained on multiple youtube comments](https://stats.stackexchange.com/questions/49363/simple-machine-learning-bot-detection) or just use ChatGPT (see screenshot below)
image

Comment (whirlxd) [reply submission/npc-comment-blocker/content.js:1]: ok to both of these points 
making a chatgpt wrapper is just a bad thing to be fair and asking it for every comment would be really poor design 
for the training model , that could be an option however considering that i dont have enough data *yet* to do that 
the keyword regex matching design works effortlessly atleast for now and has a good accuracy score 
although there may be around 5% false positives aswell however its quite accurate 95% of the time and works well
i personally use it and a small group of people i met on reddit do aswell. Appreciate your suggestions however for now i believe this is the best solution while in the future ( i am in the process of collecting more spam across different languages ) a trained model could be an option.

Comment (JosiasAurel) [reply submission/npc-comment-blocker/content.js:1]: I've seen a few comments from the linked YouTube videos that seemed legit but were hidden by your extension. I'm not sure how you measured accuracy... can you provide a bit more details?

If you're concerned about usage. Perhaps collecting multiple comments at the same time and sending a single ChatGPT query could work? Or you could search for APIs that can do better classification that a simple regex matching. 

could you also link the Reddit post you did? Just genuinely interested in seeing what other people's opinions are.

Comment (whirlxd) [reply submission/npc-comment-blocker/content.js:1]: @JosiasAurel i further did some research on the stack post you recommended and it seems i am going to be doing that however running a entire model on client side is going to be resource intensive and cause issues for some people 
also i made sure to install this extension on all my school's pcs aswell and they will definately blow up if i have to run a model.
The other option being a server is not feasible to me due to pricing issues , please understand.
Although there is scope for using something like bayes classifier that approach will take slightly more time getting past the deadline of nov 30th

**Review by JosiasAurel — APPROVED** [review]
Hey that's alright. I understand the challenge there. It'll however be best here to have the extension fetch the list of keywords from like a server. The main advantage here being that you will be able to easily add new keywords without having to modify the extension source code directly. 

Would love to see an updated version of this extension in the future. That said, please fill [this form](https://airtable.com/appgtKZwZBoOeFU7O/pagyeS51zr2kC1rp0/form) to get your grant.

**Pull request merged**